### PR TITLE
Tasks related to Prototype chain and Interface

### PR DESCRIPTION
**Pull request by Abhinav4952**
All test cases are passed.

### CODE REVIEW

File: JSTraining/Day5/interfacedemo.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/*
+We need to implement a method which checks for this interface.
+
+Rules
+The SantaClausable interface is implemented, if all of the following methods are defined on an object
+*/
+
+function isSantaClausable(obj) {
+ return (
+   (typeof obj.sayHoHoHo==='function')
```
Comment (vishal8885) [inline JSTraining/Day5/interfacedemo.js]: try using Object.getOwnPropertyNames() in this usecase

File: JSTraining/Day5/objectreference.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+You are given a complex object that has many deeply nested variables. You don't want to go the usual if obj.property == null route. 
+Create a prototype method that given a nested path, either return the value or undefined.
+*/
+// return the nested property value if it exists,
+// otherwise return undefined
+Object.prototype.hash = function(string) {
+  let arr = string.split('.');
+  //console.log(arr);
+  let ret = this;
+  for(i = 0; i < arr.length; i++) {
```
Comment (vishal8885) [inline JSTraining/Day5/objectreference.js]: avoid using for loop

File: JSTraining/Day5/interfacedemo.js (hunk 1 above)
Comment (Abhinav4952) [reply JSTraining/Day5/interfacedemo.js]: Made use of Object.getOwnPropertyNames()

File: JSTraining/Day5/objectreference.js (hunk 2 above)
Comment (Abhinav4952) [reply JSTraining/Day5/objectreference.js]: Made us of for each

**Pull request merged**